pilot/pkg/serviceregistry/serviceentry/controller.go

```diff
@@ -408,6 +408,9 @@ func (s *Controller) serviceEntryHandler(_, curr config.Config, event model.Even
 		// Delete endpoint shards only if there are no service instances.
 		if len(s.serviceInstances.getByKey(instanceKey)) == 0 {
 			s.XdsUpdater.SvcUpdate(shard, string(svc.Hostname), svc.Attributes.Namespace, model.EventDelete)
+		} else {
```

Comment:
I think this wont work. UpdatedSvcs are already processed (SvcUpdate event is already called at L400)

Reply:
@ramaraochavali SvcUpdate is currently only incrementing the metric counters when the event is not `model.EventDelete`. Though functionally it shouldn't affect, the metric wouldn't be incremented. Let me know if that's a concern

Also the main intention of adding this to updatedSvcs is to trigger eds cache update `doEdsCacheUpdate` for the host

Reply:
should work @ramaraochavali

Reply:
>  Though functionally it shouldn't affect, the metric wouldn't be incremented. Let me know if that's a concern

Yes. This is what I meant. It would be little confusing